Auth0/CredentialsManager.swift

```diff
@@ -186,14 +186,28 @@ public struct CredentialsManager {
                                                  refreshToken: credentials.refreshToken ?? refreshToken,
                                                  expiresIn: credentials.expiresIn,
                                                  scope: credentials.scope)
-                _ = self.store(credentials: newCredentials)
-                callback(nil, newCredentials)
+                if self.willExpire(newCredentials, within: minTTL) {
+                    // TODO: On the next major add a new case to CredentialsManagerError
+                    let error = NSError(domain: "The lifetime of the renewed Access Token is less than minTTL", code: -99999, userInfo: nil)
```

Comment:
let's not use magic numbers and ensure the message gives a hint on where the error could be resolved (dashboard)

Reply:
It's a number to mean "no number" or "error without error code". I took it from other `NSError` instances on some tests.
Regarding the error message, can you please provide an alternative?

Reply:
> It's a number to mean "no number" or "error without error code"

by keeping it like that it's a "magic number". Give it a name and it's no longer one. If it's used in other places as well, then extract it into a public constant or enum and reference that instead. That way, the name describes the intent of this value.

> provide an alternative?

You can point out the alternatives to resolve this issue. For example.

"The lifetime of the renewed Access Token is less than minTTL requested. Increase the 'Token Expiration' setting of your Auth0 API in the dashboard or request a lower minTTL."

Reply:
So are you suggesting that the error should output the values of the token expiry in seconds and the minTTL requirement? I think that would make it much easier to debug for sure.

Reply:
Sure, that can help. 

In this error scenario, the issue is that the token issued lasts for the next, say, 10 seconds (expiresIn); and the dev requested a token that lasts at least 20 seconds (minTTL). 

So there are two possible solutions: 
a) request a token that lasts less than 10 seconds, the lifetime defined in the dashboard (expiresIn).
b) update the dashboard settings accordingly for the server to start issuing tokens that last at least 20 seconds (minTTL), the minimum required by the dev.

Whatever we log here as the exception message should give a hint or guide to those solutions.

Reply:
I updated the error message with @lbalmaceda suggestion and added the token expiry in seconds and minTTL for easier debugging in https://github.com/auth0/Auth0.swift/pull/399/commits/04ffa990458113d1f2963b18bb82d2fc82d062c2.
Regarding the error code, it's a one-off because that particular code is not used anywhere else on the SDK target, only on the test target, and we shouldn't really be using `NSError` anyway. We're only using it because adding a new error case would be a breaking change.